resources/assets/js/components/ApplicantProfile/BasicInfo/ProfileBasicInfo.tsx

```diff
@@ -0,0 +1,495 @@
+              <Select.Option key={id} value={id}>
+                {intl.formatMessage(citizenshipDeclarationMessages(id))}
+              </Select.Option>
+            ),
+          )}
+        </Select>
+        <Select
+          name="veteranStatus"
+          required
+          register={register}
+          label={intl.formatMessage(messages.veteranStatusLabel)}
+          errorMessage={errors.veteranStatus?.message}
+          data-h2-padding="b(right, 5)"
+          data-h2-margin="b(bottom, 2)"
+        >
+          {Object.values(VeteranId).map(
+            (id: number): React.ReactElement => (
+              <Select.Option key={id} value={id}>
+                {intl.formatMessage(veteranStatusMessages(id))}
+              </Select.Option>
+            ),
+          )}
+        </Select>
+        <h2 data-h2-font-size="b(h3)" data-h2-margin="b(bottom, 1)">
+          {intl.formatMessage(messages.govtJobInfoHeading)}
+        </h2>
+        <Select
+          name="gcEmployeeStatus"
+          required
+          register={register}
+          label={intl.formatMessage(messages.gcEmployeeStatusLabel)}
+          errorMessage={errors.gcEmployeeStatus?.message}
+          data-h2-padding="b(right, 5) b(bottom, 1)"
+          onChange={(e: React.ChangeEvent<HTMLSelectElement>): void => {
+            setCurrentEmployeeStatus(parseInt(e.target.value, 10));
+          }}
+        >
+          {Object.values(GCEmployeeStatus).map((id) => (
+            <Select.Option key={id} value={id}>
+              {intl.formatMessage(gcEmployeeStatusMessages(id))}
+            </Select.Option>
+          ))}
+        </Select>
+        {/* If the user is currently a member of the GOC then allow setting a current classification */}
+        {currentEmployeeStatus === GCEmployeeStatus.current && (
+          <div data-h2-grid="b(top, expanded, flush, 1)">
+            <p data-h2-grid-item="b(1of1)">
+              {intl.formatMessage(messages.currentClassificationAndLevel)}
+            </p>
+            <Select
+              name="currentClassification"
+              defaultValue={`${currentClassification}`}
+              required
+              register={register}
+              label={intl.formatMessage(messages.classificationLabel)}
+              errorMessage={errors.currentClassification?.message}
+              data-h2-grid-item="b(1of2)"
+              data-h2-padding="b(right, 5)"
+            >
+              {classifications.map((classification) => (
+                <Select.Option
+                  key={classification.key}
+                  value={classification.id}
+                >
+                  {localizeFieldNonNull(locale, classification, "name")}
+                </Select.Option>
+              ))}
+            </Select>
+            <Select
+              name="currentLevel"
+              defaultValue={`${currentLevel}`}
+              required
+              register={register}
```

Comment:
It says to call is as a function when there are no validation rules. But since we are removing the unnecessary validation you mentioned below this will called as a func now